app/Services/RouteSuggester.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+namespace App\Services;
+
+use App\Models\Venue;
+use Fisharebest\Algorithm\Dijkstra;
+use Illuminate\Database\Eloquent\Builder;
+
+class RouteSuggester
+{
+    private float $searchRange = 0.2;
+
+    public function __construct(
+        private array $stopsAttributes,
+        private array $start,
+        private array $end
+    ) {
+    }
+
+    public function suggest()
+    {
+        $stopsVenues =
+            $this->getLocalVenuesByAttributes(
+                $this->stopsAttributes,
+                $this->start,
+                $this->end,
+                $this->searchRange
+            );
+
+        $venuesGraph = $this->createVenuesGraph($stopsVenues);
+        $dijkstra = new Dijkstra($venuesGraph);
+        $path = $dijkstra->shortestPaths('start', 'end');
+
+        if (count($path) > 0) {
+            $ids = $path[0];
+            unset($ids[0]);
+            unset($ids[count($ids)]);
+
+            $venues = Venue::whereIn('id', $ids)->get();
+            return $venues;
+        } else return false;
+    }
+
+    private function getLocalVenuesByAttributes(array $stops, array $start, array $end, float $searchRange)
+    {
+        $lat = $this->sort($start[0], $end[0]);
+        $long = $this->sort($start[1], $end[1]);
+
+        $venues = [];
+        //for each stop -> get venue ids where attributes
+        foreach ($stops as $attributes) {
```

Comment:
interesting naming here, I would argue stops as stop.

Reply:
hmm I do see what you mean, maybe stopAttributes is best.